ext/rugged/rugged_diff.c

```diff
@@ -197,6 +198,20 @@ static int diff_print_cb(
 	return GIT_OK;
 }
 
+struct nogvl_diff_patch_args {
+	git_diff * diff;
+	git_diff_format_t format;
+	VALUE rb_str;
+};
+
+static void rb_git_diff_patch_nogvl(void * _args)
+{
+	struct nogvl_diff_patch_args * args;
+
+	args = (struct nogvl_diff_patch_args *)_args;
+	git_diff_print(args->diff, args->format, diff_print_cb, (void*) args->rb_str);
```

Comment:
`diff_print_cb` calls `rb_str_cat`, and it's not safe to call `rb_str_cat` while the GVL is unlocked.  `rb_str_cat` can possibly resize the underlying string buffer, which can cause the GC to execute.  The GC is not thread safe, so this could cause havoc on the mutator.

Reply:
https://github.com/ruby/ruby/blob/8b3774be3dd9f472bddd99e84e3c9fe2ff99d7ac/thread.c#L1548-L1551 says reallocations should be same as they acquire the GVL when needed.